File: ABC/ABC301-400/ABC338/E/stack/E.py
from collections import deque
N = int(input())
P = [0] * (2 * N)
for i in range(N):
	A, B = map(int, input().split())
	A -= 1
	B -= 1
	if A > B:
		A, B = B, A
	P[A] = i
	P[B] = N + i
st = deque([])
for i in P:
	if i < N:
		st.append(i)
	else:
		if st.pop() != i - N:
			print("Yes")
			exit()
print("No")

File: ABC/ABC301-400/ABC338/E/stack/E.nim
import strutils, sequtils

let N = stdin.readLine.parseInt
var P = newSeq[int](2 * N)
for i in 0..<N:
  let a_b = stdin.readLine.split().map(parseInt)
  var
    A = a_b[0] - 1
    B = a_b[1] - 1
  if A > B:
    swap(A, B)
  P[A] = i
  P[B] = N + i
var st = newSeq[int]()
for i in P:
  if i < N:
    st.add(i)
  else:
    if st.pop() != i - N:
      echo "Yes"
      quit()
echo "No"